pytorch_lightning/loggers/comet.py

```diff
@@ -185,7 +185,8 @@ def experiment(self):
         if self._experiment is not None:
             return self._experiment
 
-        if self._future_experiment_key is not None:
+        manual_set_experiment_key_environment_variable = self._future_experiment_key is not None
+        if manual_set_experiment_key_environment_variable:
```

Comment:
This looks good, we can also move the cleaning of `self._future_experiment_key` in the finally block, this way we don't need an additional variable.

Reply:
Do you mean `self._future_experiment_key =  os.environ.pop("COMET_EXPERIMENT_KEY")` in finally or something else? I think I don't really follow your idea

Reply:
No I mean remote the line `self._future_experiment_key = None` from here and replace the finally block by:

```
            if self._future_experiment_key is not None:
                os.environ.pop("COMET_EXPERIMENT_KEY")
                self._future_experiment_key = None
```